cpg-core/specifications/eog.md

```diff
@@ -0,0 +1,733 @@
+# Specification: Evaluation Order Graph
+
+The Evaluation Order Graph (EOG) is built as edges between AST nodes after the initial translation of the code to the CPG.
+Its purpose is to follow the order in which code is executed, similar to a CFG, and additionally differentiate on a finer level of granularity in which order expressions and subexpressions are evaluated.
+Every node points to a set of previously evaluated nodes (`prevEOG`) and nodes that are evaluated after (`nextEOG`).
+The EOG-Edges are intraprocedural and thus differentiate from INVOKES-Edges.
+In the following, we summarize in which order the root node representing a language construct and its descendants in the AST tree are connected.
+
+An EOG always starts at the header of a method/function or record that holds code and ends in one (implicit) or multiple return statements.
```

Comment:
The header?

Reply:
Good point, technically its the function root node.